opmd_viewer/openpmd_timeseries/data_reader/field_reader.py

```diff
@@ -173,90 +211,52 @@ def read_field_circ( filename, field_path, m=0, theta=0. ):
-    # Extract the dataset and and corresponding group
-    group, dset = find_dataset( dfile, field_path )
-
-    # Dimensions of the grid
-    shape = list( get_shape( dset ) )
+    axis_labels = ['r', 'z']
+    shape = [Nr, Nz]
     grid_spacing = list( group.attrs['gridSpacing'] )
     global_offset = list( group.attrs['gridGlobalOffset'] )
-    # Slice selection
-    if slicing is not None:
-        # Get the integer that correspond to the slicing direction
-        slicing_index = axis_labels.index(slicing_dir)
-        # Number of cells along the slicing direction
-        n_cells = shape[ slicing_index ]
-        # Index of the slice (prevent stepping out of the array)
-        i_cell = int( 0.5 * (slicing + 1.) * n_cells )
-        i_cell = max( i_cell, 0 )
-        i_cell = min( i_cell, n_cells - 1)
-        # Remove metainformation relative to the slicing index
-        shape.pop( slicing_index )
-        grid_spacing.pop( slicing_index )
-        global_offset.pop( slicing_index )
-        new_labels = axis_labels[:slicing_index] + \
-            axis_labels[slicing_index + 1:]
-        axes = { i: new_labels[i] for i in range(len(new_labels)) }
-        # Extraction of the data
-        F = get_data( dset, i_cell, slicing_index )
-        info = FieldMetaInformation( axes, shape, grid_spacing, global_offset,
-                group.attrs['gridUnitSI'], dset.attrs['position'] )
-    else:
-        F = get_data( dset )
-        axes = { i: axis_labels[i] for i in range(len(axis_labels)) }
-        info = FieldMetaInformation( axes, F.shape,
-            group.attrs['gridSpacing'], group.attrs['gridGlobalOffset'],
-            group.attrs['gridUnitSI'], dset.attrs['position'] )
+
+    # Perform slicing if needed
+    if slicing_dir is not None:
+        # Convert arguments to lists
+        if not isinstance(slicing, list):
+            slicing = [slicing]
+        if not isinstance(slicing_dir, list):
+            slicing_dir = [slicing_dir]
+        # Slice field and clear metadata
+        # First z direction
+        if 'z' in slicing_dir:
+            ind = slicing_dir.index('z')
+            n_cells = shape[ 1 ]
+            i_cell = int( 0.5 * (slicing[ind] + 1.) * n_cells )
+            i_cell = max( i_cell, 0 )
+            i_cell = min( i_cell, n_cells - 1)
+            F_total = F_total[:, i_cell]
+            shape.pop( ind )
+            grid_spacing.pop( ind )
+            global_offset.pop( ind )
+            axis_labels = axis_labels[:-1]
+        # Then r direction
+        if 'r' in slicing_dir:
+            ind = slicing_dir.index('r')
+            n_cells = shape[ 0 ]
+            i_cell = int( 0.5 * (slicing[ind] + 1.) * n_cells )
+            i_cell = max( i_cell, 0 )
+            i_cell = min( i_cell, n_cells - 1)
+            F_total = F_total[i_cell, ...]
+            shape.pop( ind )
+            grid_spacing.pop( ind )
+            global_offset.pop( ind )
+            axis_labels = axis_labels[1:]
```

Comment:
Could you generalize the above code (up to `if 'z' in slicing_dir:`) with a loop over `slicing_dir` (just like in the cartesian case)? 

For the slicing, I think you could do 
```
F_total = np.take( F_total, i_cell, axis=ind )
```